components/home/Chat.tsx

```diff
@@ -160,22 +173,26 @@ export default function Chat() {
                     previousMessage &&
                     previousMessage.profiles.id === value.profiles.id
                   }
+                  deletePerms={
+                    (userPerms & ChannelPermissions.MANAGE_MESSAGES) !== 0
+                  }
                 />
               );
             })}
           <div ref={newestMessageRef} className=""></div>
         </div>
-
-        <MessageInput
-          onSubmit={async (content: string) => {
-            createMessage(supabase, {
-              content,
-              channel_id: channelId,
-              profile_id: user!.id,
-            });
-          }}
-        />
       </div>
+      <div className="flex grow"></div>
+      <MessageInput
+        onSubmit={async (content: string) => {
+          createMessage(supabase, {
+            content,
+            channel_id: channelId,
+            profile_id: user!.id,
+          });
+        }}
+        disabled={(userPerms & ChannelPermissions.SEND_MESSAGES) == 0}
```

Comment:
I'm leaving this to opinion, but due to the fact false-y values are considered False (0 being one of them), do we prefer for clarity what you have here?

Or would `!(userPerms & ChannelPermissions.SEND_MESSAGES)` suffice? I have no opinions either way, just whatever we all feel is most readable at a glance for what the code is trying to do

Reply:
!(userPerms & ChannelPermissions.SEND_MESSAGES) makes a lot more sense, refactored to this